### DISCUSSION TITLE
Fail on not reproducible build setup

### DISCUSSION THREAD

**Discussion by hendrikebbers** [support-and-care/maven-support-and-care#40] [Ideas for Funding]
Today most projects that use Maven does not specify a specific version for every Maven plugin that is used. This can end in different plugin versions used on different systems. For a real reproducible build the same plugin versions need to be used on every system. Based on that it would make sense to add a command line flag to maven that let a maven call fail of a plugin is used that is not specified by a fixed version. For sure that should be not active by default.

**Comment by rmannibucau** [comment]
What's the issue using the related maven plugin?
Not sure of the value to have it built-in since a lot of builds must not be reproducible by design.

**Reply by hendrikebbers** [reply]
Agree. But if you need to be reproducible (what becomes true for more and more projects based on security issues) it would be good to have a solution for that. Or do you assume that can be easily done by a plugin?

**Reply by rmannibucau** [reply]
I assume the plugin is already there between artifact plugin (https://maven.apache.org/guides/mini/guide-reproducible-builds.html) and the fact we warn today (to not break backward compat) when plugins are not pinned (we want to keep defaulting on hardcoded one for now).
So if you want it to fail you make warning failing the build (which is generally good) but feature is there already IMHO.

**Comment by ascheman** [comment]
This discussion has been converted to Epic #67 for tracking.

Closing this discussion.
